src/mapping/Mapping.cpp

```diff
@@ -127,8 +127,10 @@ bool operator<(Mapping::MeshRequirement lhs, Mapping::MeshRequirement rhs)
   case (Mapping::MeshRequirement::UNDEFINED):
     return rhs != Mapping::MeshRequirement::UNDEFINED;
   case (Mapping::MeshRequirement::VERTEX):
-    return rhs == Mapping::MeshRequirement::FULL;
+    return rhs == Mapping::MeshRequirement::FULL || rhs == Mapping::MeshRequirement:: GRADIENT;
   case (Mapping::MeshRequirement::FULL):
+    return rhs == Mapping::MeshRequirement::GRADIENT;
```

Comment:
I assumed that the order in this case is not very relevant, it needs to be mainly consistent everywhere in the code. For now the enums are given an integer value (1 for VERTEX, 2 for FULL and 3 for GRADIENT) so this was the order I used. But indeed, it makes sense to assume the FULL Requirement would be ranked higher.  Should I also change their respective integer values?

Reply:
> I assumed that the order in this case is not very relevant, it needs to be mainly consistent everywhere in the code.

No, the order and integer values are not relevant (and you don't need to change it). I just said 'lower' since its actually the 'lower' operator you define here (`operator<`).